### ISSUE TITLE
Nuget Documentation

### ISSUE THREAD

**Issue by splendid25** [tjscience/RoboSharp#73]
Hi,

There is no documentation in the nuget package, but I see the source code is well documented.
I think it's just a matter of embedding the XML in the nuget ?

**Comment by RFBomb** [comment]
I too have noted this issue before. 

This thread has the fix for it: 

https://stackoverflow.com/questions/43305578/show-comments-in-nuget-package

https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2010/x4sa0ak0(v=vs.100)?redirectedfrom=MSDN

It seems to be just a simple checkbox isn’t being ticked in the compiler options page.

**Comment by PCAssistSoftware** [comment]
Invalid path for XML documentation file in RoboSharp.csproj - it is looking for it in "C:\Users\rbrenckman\source\repos\RobotSoftwareTool_.Net Framework\RoboSharp\RoboSharp\RoboSharp.xml"

**Comment by RFBomb** [comment]
Yea that’s my bad. I tried fixing that multiple times by making it
relative, but it wasn’t sticking. I had hoped that publishing it into the
nugget package would bundle it into the assembly.

The ‘Build Action’ property on the file likely has to be changed to
‘embedded resource’. Also, look at ‘Copy To Output Directory’

**Comment by PCAssistSoftware** [comment]
Now looking... also think the below needs adding as per one of the above articles to csproj file as it seems using the checkbox makes it an absolute path and causes the issue

` <GenerateDocumentationFile>true</GenerateDocumentationFile>`

**Comment by PCAssistSoftware** [comment]
Okay, it does seem that if you check the box "XML documentation file" on the build tab then for some reason it wrongly creates an absolute path as mentioned here - https://stackoverflow.com/questions/43305578/show-comments-in-nuget-package and here - https://developercommunity.visualstudio.com/t/default-xml-documentation-file-path-is-absolute/319886.

The solution is to NOT check the box but to manually edit the RoboSharp.csproj file and add the below:-

```
<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net40|AnyCPU'">
    	<GenerateDocumentationFile>true</GenerateDocumentationFile>
  </PropertyGroup>
```

This seems to work as RoboSharp.XML is correctly copied to \RoboSharp-dev\RoboSharp-dev\RoboSharp\bin\Debug\net40 when building

Will update files manually now, @tjscience - you may need to do whatever steps you do to update the NuGet project

**Comment by PCAssistSoftware** [comment]
Not sure we need to stipulate Condition of net40 only for the documentation e.g. 

`<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net40|AnyCPU'">`

As surely the documentation XML file is applicable for all builds (net40, net45, netcoreapp3.1, netstandard2.0 and netstandard2.1)

Edit: Tested in all builds and can't see reason for condition so removed

**Comment by RFBomb** [comment]
@PCAssistSoftware 
I just did some testing with the 1.2.5 version of the nuget package. It seems that enabling this may not be enough, as I was unable to see the xml documentation when hovering over the robosharp object's properties. (For example, hovering on RoboSharpObject.LoggingOptions provided no tooltip)

From my testing of compiling it myself, what I've found is that the XML documentation is copied to each of the output folders (net40, net45, etc). Deleting that xml file then referencing the DLL causes no tooltips. Referencing the DLL while the XML file exists in the same folder will enable the tooltips, even if the xml file is later deleted from that folder. (thats weird, maybe VS is importing and storing it somewhere in the project, but whatever, thats besides the point)

I inspected the self-compiled package and compared it to the package on the NuGet website. RoboSharp.XML is missing from the 1.2.5 version on available via nuget. More specifically, the xml file only exists in the Net40 version, and is missing in all others.

I was able to compile a few time (once or twice) and saw it missing from the other folders. But, more and more testing its showing up in all the folders like its supposed to. So I'm a bit confused as to whats happening there. It may be an artifact of when I did the initial change to generate it, I'm unsure, after which the fix to the dev repo was made.

Should `obj\Release\net40\RoboSharp.xml` path be something else? maybe simply `obj\Release\RoboSharp.xml` ? I can't tell if this is a contributor or not. (Having it just as RoboSharp.xml will cause it to copy into the source directly, and consequently appear in all Pull Requests since the file is regenerated every compile, so having it in one of the ignored folders makes sense)

But, TLDR: the xml file is missing from all versions except net40 in the 1.2.5 nuget package that was uploaded.

**Comment by PCAssistSoftware** [comment]
the published version 1.2.5 in NuGet DOESN'T include the later change I made 3 days ago so no I wouldn't expect it to work - as this version is one published by @tjscience BEFORE I found the issue and fixed it - hence why above I asked if he could republish it to NuGet after I had made change....

It should then work fine, as having RoboSharp.DLL and RoboSharp.XML in same location (copied manually for testing) shows the documentation correctly in VS

**Comment by RFBomb** [comment]
Yea, thats what I found as well. Just wasn't sure if this was caught. For now, I'm just going to copy the xml file into my packages... folder where needed.
